### ISSUE TITLE
Can't cast NoopNode to ClockNode on Android

### ISSUE THREAD

**Issue by likern** [software-mansion/react-native-reanimated#672]
I use animation function, which is defined as follows:
```typescript
const animateTranslation = (
  clock: Animated.Clock,
  position: Animated.Value<number>,
  translation: Animated.Value<number>,
  offset: Animated.Value<number>,
  min: Animated.Value<number>,
  max: Animated.Value<number>,
  gestureState: Animated.Value<State>
) => {
  const snapPoint = new Value(0);
  const myclock = new Clock();
  return cond(clockRunning(myclock), snapPoint, snapPoint);
};
```
and later use the returned value to animate `Animated.View's` property:
```typescript
const taskTranslateY = animateTranslation(clock, position, translationY, offsetY, min, max, state);
...
<Animated.View
  style={{
    flex: 1,
    flexDirection: 'column',
    marginTop: -200,
    transform: [{ translateY: taskTranslateY }]
  }}
>
...
</Animated.View>
```
And I started getting this error, which I can't understand or somehow to deal with. Where is a bug?
![Снимок экрана от 2020-03-21 20-37-25](https://user-images.githubusercontent.com/842306/77232708-cc5a2800-6bb3-11ea-8f1e-e67f9e7b5bab.png)

**Comment by likern** [comment]
I think the problem is when we deal with clock and `clockRunning, startClock, stopClock` functions.
I could catch this error just by using 
```ts
debug('clock before timing? ', clockRunning(clock))
```

**Comment by jakub-gonet** [comment]
Hi @likern,
could you post the entire code, especially everything that happens with `clock`? This is related to casting `Node` to `ClockNode` in java native code: `return ((ClockNode) clock).isRunning ? 1. : 0.;` in class `ClockTestNode`, which is code evaluating `clockRunning` node.

**Comment by jakub-gonet** [comment]
Closing due to no activity.

**Comment by Ashwin-Mothilal** [comment]
I had the same issue when I have initialised a Clock and trying to stop it without actually starting the Clock. Will this help you @jakub-gonet ?

**Comment by jakub-gonet** [comment]
@Ashwin-Mothilal,

no, I couldn't reproduce it that way.

Check it yourself:
```js
import React from 'react';
import { TextInput } from 'react-native';
import Animated from 'react-native-reanimated';

const {
  useCode,
  concat,
  Clock,
  stopClock,
  createAnimatedComponent,
} = Animated;

const AnimText = createAnimatedComponent(TextInput);

const Example = () => {
  const clock = new Clock();
  const value = new Value(10);
  useCode(() => [stopClock(clock)], []);
  return <AnimText editable={false} text={concat('', clock)} />;
};

export default Example;
```

**Comment by Ashwin-Mothilal** [comment]
@jakub-gonet I didn't get the error for your code but I replaced the return part with the below, I was able to reproduce it.

```
return (
    <Animated.View
      style={{
        height: 100,
        width: 100,
        backgroundColor: 'red'
      }}
    />
  );
```

**Comment by lightrow** [comment]
had similar issue and it seems wrapping the clock with useRef resolved it.

` const clock = useRef(new Clock());`

**Comment by pearup** [comment]
I'm having the same issue, clock is basically unusable on android,
Basically anything I do with clock will trigger the same issue.
Using 1.7.0

**Comment by wcandillon** [comment]
Thanks @pearup that fixed the issue for me.

**Comment by avinashlng1080** [comment]
@pearup  can you please paste a snippet of your solution here ?

**Comment by hosseinmd** [comment]
@pearup Timing doesn't have velocity. Where do you define velocity?

**Comment by luogao** [comment]
I add a cond block to check whether the clock is defined. 
and it worked for me
``` javascript
          cond(
              defined(this.clock),
              [
                cond(
                  not(clockRunning(this.clock)),
                  [ set(this.startValue, 1) ],
                  0
                )
              ],
              0
            )

```

**Comment by Lg0gs** [comment]
@luogao Thank, you solved my problem..

**Comment by gabrielbull** [comment]
For me, this bug occurred on a large FlatList with many items and towards the bottom of the list, the bug would always happened. The hack that worked to circumvent this bug for me was to disable anything that would call the clocks on the first render and enable on the second. Since I was using PanGestureHandler, this did the trick:

```typescript
  const [enabled, setEnabled] = useState(false)
  useLayoutEffect(() => {
    setEnabled(true)
  }, [])
  
  return <PanGestureHandler enabled={enabled} />
```

Not ideal since it triggers an extra re-render, but it does avoid this bug.

**Comment by jakub-gonet** [comment]
@gabrielbull could you share your example so I can test it as well?

**Comment by piaskowyk** [comment]
The inactive issue so I close. If anyone has still this problem, please let me know or try Reanimated v2.

**Comment by rtauziac** [comment]
I did have this very issue but with different settings.

I was starting an animation with different lengths depending on the `pressed` state of a `Pressable` so the button scales instantly on `pressDown` and animates on `pressUp`.

```typescript
Animated.timing(fadeRightButtonAnim, {
  toValue: pressed ? 1.2 : 1,
  duration: pressed ? 0 : 200,
  easing: EasingNode.out(EasingNode.exp),
}).start();
```

It worked on iOS but crashes with the error **Can't cast NoopNode to ClockNode** on Android.

The fix I applied was to put an animation length greater than zero.

```typescript
  duration: pressed ? 1 : 200, // changed from 0 to 1
```

Hope this can help someone ✨
